packages/0xcert-ethereum-gateway-contracts/src/contracts/actions-gateway.sol

```diff
@@ -403,167 +406,89 @@ contract ActionsGateway is
    * @param _order Data needed for order.
    */
   function _doActionsReplaceZeroAddress(
-    OrderData memory _order
+    OrderData memory _order,
+    address _replaceAddress
   )
     private
   {
     for(uint256 i = 0; i < _order.actions.length; i++)
     {
       require(
-        proxies[_order.actions[i].proxy] != address(0),
+        proxies[_order.actions[i].proxyId].proxyAddress != address(0),
         INVALID_PROXY
       );
 
-      if (_order.actions[i].kind == ActionKind.create)
+      if (proxies[_order.actions[i].proxyId].kind == ActionKind.create)
       {
         require(
-          Abilitable(_order.actions[i].token).isAble(_order.maker, ABILITY_ALLOW_CREATE_ASSET),
+          Abilitable(_order.actions[i].contractAddress).isAble(
+            _order.signers[bytesToUint8(85, _order.actions[i].params)],
```

Comment:
Code sniff: no magic numbers in code please. Please use constants at the top of the file.